### PR TITLE
Use Tanstack data in Overview AddressesList and addresses' filtering

### PR DESCRIPTION
**Pull request by nop33**

### CODE REVIEW

File: apps/desktop-wallet/src/components/AddressRow.tsx [hunk 1]
```diff
@@ -16,33 +16,32 @@ You should have received a copy of the GNU Lesser General Public License
...
-  address: Address
-  disableAddressCopy?: boolean
-  onClick?: (address: Address) => void
+  addressHash: AddressHash
+  onClick?: (addressHash: AddressHash) => void
   className?: string
 }
 
-const AddressRow: FC<AddressRowProps> = ({ address, disableAddressCopy, onClick, children, className }) => (
+const AddressRow: FC<AddressRowProps> = ({ addressHash, onClick, children, className }) => (
```
Comment (mvaivre) [inline apps/desktop-wallet/src/components/AddressRow.tsx:32]: nit: I know it's not part of this PR, but should we remove `FC` for consistency sake?

File: apps/desktop-wallet/src/features/addressFiltering/useFilteredAddressHashes.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            }`.toLowerCase()
+
+            return (
+              address &&
+              (address.label?.toLowerCase().includes(searchString) ||
+                address.hash.toLowerCase().includes(searchString) ||
+                addressAssetNamesWithBalances.includes(searchString))
+            )
+          })
+    const filteredByToggle = hideEmpty
```
Comment (mvaivre) [inline apps/desktop-wallet/src/features/addressFiltering/useFilteredAddressHashes.tsx:89]: Nit: space above

**Review by mvaivre — COMMENTED** [review]
Part 1

File: apps/desktop-wallet/src/pages/UnlockedWallet/AddressesPage/AddressesTabContent.tsx [hunk 3]
```diff
@@ -80,9 +67,7 @@ const AddressesTabContent = () => {
     >
       <TableGrid>
         <TableGridContent>
-          {visibleAddresses.map((address) => (
-            <AddressGridRow addressHash={address.hash} key={address.hash} />
-          ))}
+          {visibleAddresses?.map((addressHash) => <AddressGridRow addressHash={addressHash} key={addressHash} />)}
```
Comment (mvaivre) [inline apps/desktop-wallet/src/pages/UnlockedWallet/AddressesPage/AddressesTabContent.tsx:70]: What about adding a placeholder text here if no address is found?

**Comment by mvaivre** [comment]
A few tiny comments. Great job, it works well!

File: apps/desktop-wallet/src/components/AddressRow.tsx (hunk 1 above)
Comment (nop33) [reply apps/desktop-wallet/src/components/AddressRow.tsx:32]: We have an old issue for it: https://github.com/alephium/alephium-frontend/issues/75

File: apps/desktop-wallet/src/pages/UnlockedWallet/AddressesPage/AddressesTabContent.tsx (hunk 3 above)
Comment (nop33) [reply apps/desktop-wallet/src/pages/UnlockedWallet/AddressesPage/AddressesTabContent.tsx:70]: 💯

File: apps/desktop-wallet/src/components/AddressRow.tsx (hunk 1 above)
Comment (mvaivre) [reply apps/desktop-wallet/src/components/AddressRow.tsx:32]: Yep! I thought we said that we can make the changes as we go, when we stumble upon it, but I may be mistaken. Anyway, just a nit!

File: apps/desktop-wallet/src/pages/UnlockedWallet/AddressesPage/AddressesTabContent.tsx (hunk 3 above)
Comment (nop33) [reply apps/desktop-wallet/src/pages/UnlockedWallet/AddressesPage/AddressesTabContent.tsx:70]: Done 
image

File: apps/desktop-wallet/src/features/addressFiltering/useFilteredAddressHashes.tsx (hunk 2 above)
Comment (nop33) [reply apps/desktop-wallet/src/features/addressFiltering/useFilteredAddressHashes.tsx:90]: Done

File: apps/desktop-wallet/src/api/addressQueries.ts [hunk 4]
```diff
@@ -40,15 +45,27 @@ export const addressAlphBalanceQuery = ({
...
+        addressHash,
+        balances
+      }
+    },
     placeholderData: queryClient.getQueryData([
       ...ADDRESS_BALANCE_QUERY_KEYS,
       'ALPH',
       { addressHash, latestTxHash: previousTxHash, networkId }
-    ]) as AddressBalance,
+    ]) as AddressAlphBalanceQueryResult,
```
[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/api/addressQueries.ts:60]: That's a very good point, thanks for this. I'll investigate.

[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/api/addressQueries.ts:60]: What do you think about this? No need to type cast anymore:

```ts
export const addressAlphBalanceQuery = ({
  addressHash,
  networkId,
  latestTxHash,
  previousTxHash
}: AddressBalanceQueryProps) => {
  const getQueryOptions = (latestTxHash: AddressBalanceQueryProps['latestTxHash']) =>
    queryOptions({
      queryKey: [...ADDRESS_BALANCE_QUERY_KEYS, 'ALPH', { addressHash, latestTxHash, networkId }],
      queryFn: async () => {
        const balances = await client.explorer.addresses.getAddressesAddressBalance(addressHash)

        return {
          addressHash,
          balances
        }
      },
      staleTime: Infinity
    })

  const previousQueryKey = getQueryOptions(previousTxHash).queryKey
  const latestQueryOptions = getQueryOptions(latestTxHash)

  return queryOptions({
    ...latestQueryOptions,
    placeholderData: queryClient.getQueryData(previousQueryKey)
  })
}
```

I could make the code shorter, but I find it more visually beautiful and readable this way.

[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/api/addressQueries.ts]: Pushed

[the remark this answers is not in the document]
Comment (mvaivre) [reply apps/desktop-wallet/src/api/addressQueries.ts]: This LGTM! 👏

**Review by mvaivre — APPROVED** [review]
LGTM!

**Pull request merged**